### PR TITLE
refactor fix/backend 

### LINKED ISSUE #9: Refactor Models to reduce hard coding
Models(specifically Laptime) have a lot of repetitive code.
```python
#TRACK_NOW mongo models
class Car:
    def __init__(self, id, name, body, car_class, engine, hp, layout, racecar, transmission):
        self.id = id
        self.name = name
        self.body = body
        self.car_class = car_class
        self.engine = engine
        self.hp = hp
        self.layout = layout
        self.racecar = racecar
        self.transmission = transmission

    def to_dict(self):
        return {
            'id': self.id,
            'name': self.name,
            'body': self.body,
            'car_class': self.car_class,
            'engine': self.engine,
            'hp': self.hp,
            'layout': self.layout,
            'racecar': self.racecar,
            'transmission': self.transmission
        }

# Laptime Model
class Laptime:
    def __init__(self, car_id, track_id, driver_id, time, date):
        self.car_id = car_id
        self.track_id = track_id
        self.driver_id = driver_id
        self.time = time
        self.date = date

    def to_dict(self):
        return {
            'car_id': self.car_id,
            'track_id': self.track_id,
            'driver_id': self.driver_id,
            'time': self.time,
            'date' : self.date
        }
    def only_tracks(self):
        return {
            'track_id': self.track_id
        }
    def only_drivers(self):
        return {
            'driver_id': self.driver_id
        }
    def only_cars(self):
        return {
            'car_id' : self.car_id
        }
# Driver Model
class Driver:
    def __init__(self,id, name, nationality):
        self.id = id
        self.name = name
        self.nationality = nationality

    def to_dict(self):
        return {
            'id': self.id,
            'name': self.name,
            'nationality': self.nationality
        }


```
car_id and driver_id is being hardcoded whereas it could inherit from The Driver Model and Car class

### PR DESCRIPTION
**Pull request by Dozie2001**
Closes #9. This refactor offers a cleaner  approach for Models

### CODE REVIEW

File: DEPENDENCIES.txt [hunk 1]
```diff
@@ -7,4 +7,5 @@ itsdangerous==2.1.2
 Jinja2==3.1.2
 MarkupSafe==2.1.3
 pymongo==4.6.1
```
Comment (Dozie2001) [inline DEPENDENCIES.txt:9]: You didn't add this to the DEPENDENCIES

File: backend/models.py [hunk 2]
```diff
@@ -24,20 +24,42 @@ def to_dict(self):
...
         self.time = time
         self.date = date
 
     def to_dict(self):
+        dict_car = Car.to_dict(self)
+        dict_Driver = Driver.to_dict(self)
         return {
-            'car_id': self.car_id,
+            'car_id': dict_car['id'],
             'track_id': self.track_id,
```
Comment (vantage-ola) [inline backend/models.py:61]: you would have to do for track_id also, so dict_track['id']...

**Review by vantage-ola — CHANGES_REQUESTED** [review]
good, just need to do the same for track_id

File: backend/models.py [hunk 3]
```diff
@@ -83,4 +55,42 @@ def to_dict(self):
...
+        return {
+            'car_id': dict_car['id'],
+            'track_id': dict_Track['id'],
+            'driver_id':  dict_Driver['id'],
+            'time': self.time,
+            'date' : self.date
+        }
+    def only_tracks(self):
+        return {
+            'track_id': self.track_id
```
Comment (vantage-ola) [inline backend/models.py:87]: *self.Track.id , just this change and its fine....good work bro..

**Review by vantage-ola — CHANGES_REQUESTED** [review]

**Review by Dozie2001 — COMMENTED** [review]
I have fixed the mistakes I made

**Review by vantage-ola — APPROVED** [review]
good one bro

**Pull request merged**